modules/market/src/main/java/com/opengamma/strata/market/curve/DefaultCurveMetadataBuilder.java

```diff
@@ -0,0 +1,258 @@
+  /**
+   * The metadata about the parameters.
+   * <p>
+   * If present, the parameter metadata will match the number of parameters on the curve.
+   */
+  private List<CurveParameterMetadata> parameterMetadata;
+
+  /**
+   * Restricted constructor.
+   */
+  DefaultCurveMetadataBuilder() {
+  }
+
+  /**
+   * Restricted copy constructor.
+   * 
+   * @param beanToCopy  the bean to copy from
+   */
+  DefaultCurveMetadataBuilder(DefaultCurveMetadata beanToCopy) {
+    this.curveName = beanToCopy.getCurveName();
+    this.xValueType = beanToCopy.getXValueType();
+    this.yValueType = beanToCopy.getYValueType();
+    this.info.putAll(beanToCopy.getInfo());
+    this.parameterMetadata = beanToCopy.getParameterMetadata().map(m -> new ArrayList<>(m)).orElse(null);
+  }
+
+  //-----------------------------------------------------------------------
+  /**
+   * Sets the curve name.
+   * 
+   * @param curveName  the curve name
+   * @return this, for chaining
+   */
+  public DefaultCurveMetadataBuilder curveName(CurveName curveName) {
+    ArgChecker.notNull(curveName, "curveName");
+    this.curveName = curveName;
+    return this;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Sets the x-value type, providing meaning to the x-values of the curve.
+   * <p>
+   * This type provides meaning to the x-values. For example, the x-value might
+   * represent a year fraction, as represented using {@link ValueType#YEAR_FRACTION}.
+   * <p>
+   * If using the builder, this defaults to {@link ValueType#UNKNOWN}.
+   * 
+   * @param xValueType  the x-value type
+   * @return this, for chaining
+   */
+  public DefaultCurveMetadataBuilder xValueType(ValueType xValueType) {
+    ArgChecker.notNull(xValueType, "xValueType");
+    this.xValueType = xValueType;
+    return this;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Sets the y-value type, providing meaning to the y-values of the curve.
+   * <p>
+   * This type provides meaning to the y-values. For example, the y-value might
+   * represent a zero rate, as represented using {@link ValueType#ZERO_RATE}.
+   * <p>
+   * If using the builder, this defaults to {@link ValueType#UNKNOWN}.
+   * 
+   * @param yValueType  the y-value type
+   * @return this, for chaining
+   */
+  public DefaultCurveMetadataBuilder yValueType(ValueType yValueType) {
+    ArgChecker.notNull(yValueType, "yValueType");
+    this.yValueType = yValueType;
+    return this;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Sets the day count.
+   * <p>
+   * This stores the day count in the additional information map using the
+   * key {@link CurveInfoType#DAY_COUNT}.
+   * 
+   * @param dayCount  the day count, may be null
+   * @return this, for chaining
+   */
+  public DefaultCurveMetadataBuilder dayCount(DayCount dayCount) {
```

Comment:
The only reason I can think of for allowing a null argument is to allow reuse of an existing builder where specifying a null day count would clear the previously set day count. 

That doesn't happen here; if the day count is null the previous day count is left in place and the method returns successfully. I think the behaviour should be either of the following when the day count is null

* Throw an `IllegalArgumentException`
* `info.remove(CurveInfoType.DAY_COUNT)`

I don't have a strong opinion on which one it is.